pkg/network/port_test.go

```diff
@@ -190,3 +192,11 @@ func getPort(t *testing.T, listener net.Listener) uint16 {
 func getPortUDP(_ *testing.T, udpConn *net.UDPConn) uint16 {
 	return uint16(udpConn.LocalAddr().(*net.UDPAddr).Port)
 }
+
+func isUnusedUDPPort(_ *testing.T, port int) bool {
+	_, err := net.ListenUDP("udp", &net.UDPAddr{Port: port})
+	if err != nil && strings.Contains(err.Error(), "address already in use") {
+		return false
+	}
+	return true
```

Comment:
We should close the listener above in the case of success. You can do that by calling `t.Cleanup` here.

Reply:
Fixed.